src/App.tsx

```diff
@@ -15,36 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+export enum SortType {
+  ALPHABETICAL = 'ALPHABETICAL',
+  LENGTH = 'LENGTH',
+  REVERSE = 'REVERSE',
+  DEFAULT = 'DEFAULT',
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortedGoods = [...goodsFromServer].sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.ALPHABETICAL:
+        return good1.localeCompare(good2);
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  const displayedGoods = isReversed ? [...sortedGoods].reverse() : sortedGoods;
+
+  const handleSort = (field: SortType) => {
+    setSortField(field);
+  };
+
+  const handleReverse = () => {
+    setIsReversed(!isReversed);
```

Comment:
```suggestion
    setIsReversed(curr => !curr);
```

always use callback when you need to set state based on the previous one